src/EditorFeatures/CSharp/UseAutoProperty/UseAutoPropertyAnalyzer.cs

```diff
@@ -12,48 +14,100 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UseAutoProperty
+            List<AnalysisResult> analysisResults)
         {
-            return ((CSharpCompilation)compilation).LanguageVersion >= LanguageVersion.CSharp6;
+            if (member.IsKind(SyntaxKind.NamespaceDeclaration, out NamespaceDeclarationSyntax namespaceDeclaration))
+            {
+                AnalyzeMembers(context, namespaceDeclaration.Members, analysisResults);
+            }
+
+            // If we have a class or struct, recurse inwards.
+            if (member.IsKind(SyntaxKind.ClassDeclaration, out TypeDeclarationSyntax typeDeclaration) ||
+                member.IsKind(SyntaxKind.StructDeclaration, out typeDeclaration))
+            {
+                AnalyzeMembers(context, typeDeclaration.Members, analysisResults);
+            }
+
+            if (member is PropertyDeclarationSyntax propertyDeclaration)
+            {
+                AnalyzeProperty(context, propertyDeclaration, analysisResults);
+            }
         }
 
-        protected override void RegisterIneligibleFieldsAction(CompilationStartAnalysisContext context, ConcurrentBag<IFieldSymbol> ineligibleFields)
+        protected override void RegisterIneligibleFieldsAction(
+            List<AnalysisResult> analysisResults, HashSet<IFieldSymbol> ineligibleFields,
+            Compilation compilation, CancellationToken cancellationToken)
         {
-            context.RegisterSyntaxNodeAction(snac => AnalyzeArgument(ineligibleFields, snac), SyntaxKind.Argument);
+            var groups = analysisResults.Select(r => (TypeDeclarationSyntax)r.PropertyDeclaration.Parent)
+                                        .Distinct()
+                                        .GroupBy(n => n.SyntaxTree);
+
+            foreach (var group in groups)
+            {
+                var tree = group.Key;
+                var semanticModel = compilation.GetSemanticModel(tree);
+
+                foreach (var typeDeclaration in group)
+                {
+                    foreach (var argument in typeDeclaration.DescendantNodesAndSelf().OfType<ArgumentSyntax>())
+                    {
+                        AnalyzeArgument(semanticModel, argument, ineligibleFields, cancellationToken);
+                    }
+                }
+            }
         }
 
         protected override ExpressionSyntax GetFieldInitializer(VariableDeclaratorSyntax variable, CancellationToken cancellationToken)
         {
             return variable.Initializer?.Value;
         }
 
-        private void AnalyzeArgument(ConcurrentBag<IFieldSymbol> ineligibleFields, SyntaxNodeAnalysisContext context)
+        private void AnalyzeArgument(
+            SemanticModel semanticModel, ArgumentSyntax argument,
+            HashSet<IFieldSymbol> ineligibleFields, CancellationToken cancellationToken)
         {
             // An argument will disqualify a field if that field is used in a ref/out position.  
             // We can't change such field references to be property references in C#.
```

Comment:
❗️ Need to handle case where a struct is passed to a `ref this` extension method (either as part of this pull request, or by creating a new issue and replying to this comment with the new issue number).

Reply:
Done: https://github.com/dotnet/roslyn/issues/23257